include/deal.II/particles/particle_handler.h

```diff
@@ -717,21 +717,89 @@ namespace Particles
     void
     update_ghost_particles();
 
+    /**
+     * This function prepares the particle handler for a coarsening and
+     * refinement cycle, by storing the necessary information to transfer
+     * particles to their new cells. The implementation depends on the
+     * triangulation type that is connected to the particle handler and differs
+     * between shared and distributed triangulations. This function should be
+     * used like the corresponding function with the same name in the
+     * SolutionTransfer() class.
+     */
+    void
+    prepare_for_coarsening_and_refinement();
+
+    /**
+     * This function unpacks the particle data after a coarsening and
+     * refinement cycle, by reading the necessary information to transfer
+     * particles to their new cells. This function should be
+     * used like the SolutionTransfer::interpolate() function after mesh
+     * refinement has finished. Note that this function requires a working
+     * mapping, i.e. if you use a mapping class that requires setup after
+     * a mesh refinement (e.g. MappingQCache(), or MappingEulerian()), the
+     * mapping has to be ready before you can call this function.
+     *
+     * @note It is important to note, that if you do not call this function
+     * after a refinement/coarsening operation (and a repartitioning operation
+     * in a distributed triangulation) and the particle handler contained
+     * particles before, the particle handler will not be usable any more. Not
+     * calling this function after refinement would be similar to trying to
+     * access a solution vector after mesh refinement without first using a
+     * SolutionTransfer class transfer the vector to the new mesh.
+     */
+    void
+    unpack_after_coarsening_and_refinement();
```

Comment:
> I am not sure how many places use CellDataTransfer, but if we change the interface of SolutionTransfer, I think we have a lot of work and convincing to do :-)

I agree. `SolutionTransfer` is so important and widely used in the `deal.II` community, that changing the interface will let us face a lot of resistance. However, finding a common nameset will make it easier for new users.

> Starting from a clean slate I think I would like the combination prepare_for_mesh_transfer/unpack_after_mesh_transfer best. mesh_transfer also captures the repartition case (currently we need to prepare for coarsening and refinement even if we only do a distributed repartition), and also references the name of the classes (CellDataTransfer/SolutionTransfer). And adding _after_mesh_transfer to the unpack function may not be necessary for CellDataTransfer, because its only purpose is the transfer, but I think particle_handler.unpack() is not quite descriptive enough, while particle_handler.unpack_after_mesh_transfer() make immediate sense. Also I like the symmetry ^^.

I also agree with you on the generalization of the suffix. How about reducing it from `mesh_transfer` to `transfer` only? This way we would end up with functions `prepare_for_transfer` and `unpack_after_transfer`, which I like very much.

What do you think of calling the post transfer function `distribute_after_transfer` instead? I'm undecided whether `unpack` sounds too technical in that regard, and that we should rather use a different word here. However, the word `distribute` is kind of biased due to `DoFHandler::distribute_dofs()`, which might lead to confusion.